src/reducers/url-state.js

```diff
@@ -232,16 +232,12 @@ const timelineType: Reducer<TimelineType> = (
 ) => {
   switch (action.type) {
     case 'PROFILE_LOADED':
-      if (!action.profile.meta.categories) {
-        // An imported profile did not provide its own categories. Use the stack view instead.
-        return 'stack';
-      }
       return state;
```

Comment:
You are right about the order of the actions. But there is one problem with the `UPDATE_URL_STATE`. 
When we add it to the `LOAD_PROFILE` it changes the url state. But then, during the `UPDATE_URL_STATE`, it actually [replaces the whole urlState object with a new one](https://github.com/firefox-devtools/profiler/blob/9d09fe383cfcfb56c21013105820369aa05bb939/src/reducers/url-state.js#L675-L681). Because of this, whenever we update the url state during the `LOAD_PROFILE`, it's going to be overwritten no matter what. Because of this, we need to do it during the finalize function to make sure that we have a new url state and we can do this addition afterwards depending on the profile data. So unfortunately, we can't change it during the `LOAD_PROFILE`  action. 

I will remove this case from the reducer instead.